src/components/My-profile.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { createSelector } from 'reselect';
+import '../styles/my-profile.css';
+import { leaveMission } from '../redux/missions/missionsSlice';
+import { cancelationRocket } from '../redux/rockets/rocketsSlice';
+
+const selectMissions = (state) => state.missions.missions;
+const selectRockets = (state) => state.rockets.rockets;
+
+const selectReservedMissions = createSelector(
+  [selectMissions],
+  (missions) => missions.filter((mission) => mission.reserved),
+);
+
+const selectReservedRockets = createSelector(
+  [selectRockets],
+  (rockets) => rockets.filter((rocket) => rocket.reserved),
+);
+
```

Comment:
- [ ] Great job so far working on the project I think you made a huge effort on it and it is really appreciated but I think you have a small bug here so when displaying the missions and rockets on my profile page when I try to reserve a rocket and go to the profile page I can see it but when then come back to the rockets page I could see the reserve is gone but it should still there till the user cancel it by himself the same for missions so kindly fix it to still show that it is reserved 🙏🏻✨😄✅🔥


- Here's a video about what I am asking for ⬇


https://github.com/anyars-encarta/react-group-project/assets/58553711/514d96b8-7d5e-45c4-9419-f668b6d062c9

Reply:
Thank you for that observation. We will look into it